language-design.md

```diff
@@ -26,15 +76,45 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
+A well-designed language is one that, first and foremost, fits its purpose. A language that is 
+worse at its intended purpose than other languages is not a good language. For example, if I design 
+a statistical modeling language that only lets me draw pictures of cows, my language is not a 
+well-designed statistical modeling language (although it might be a good bovine sketching 
+language). However, just because a language fits its current purpose, it may not necessarily always 
+fit that purpose. This is what Steele is so concerned about: languages must be able to grow and 
+evolve, in part driven by their users, in order to stay relevant [Steele, 1998]. Sometimes, though, 
+too much growth can be a bad thing: languages can become bloated masses, used for things far 
+outside of their original purpose. Growth must be controlled, and focused around making the 
+language better at its specific niche. Conversely, a language should not be too small; if users 
+find themselves using the same boilerplate code or external library in every single program, 
+perhaps its time to grow the language a bit. Still, "you can always add things later, but
+you can’t take them away" [Bloch, 2006]; make sure everything added will work. Finally, a 
+well-designed language is human-centered: it must be intelligible to its users and have consistent 
+notation [Bloch, 2006].
+
+A poorly-designed language lacks one or more of these features: it fails to evolve, is too small or 
+too large for its purpose, or is nonsensical to everyone but its hardcore devotees. Symptoms of a 
+poorly-designed language are re-using your own code constantly (the language is too small), having 
+to check the documentation constantly (the language is too large or utterly unintelligible), and 
+having keywords or operators that have behavior contrary to what most programmers would expect, 
+like the Swift let keyowrd representing constants while the JavaScript let keyword representing 
+variables, or having gray(100) represent the color white [Verou, 2014].
+
```

Comment:
I like your point about a language having a purpose - it's relevant to our class and specific domains, but also a GPL has a purpose in being able to solve lots of problems in a specifically designed way. I also think your argument about expected behavior is interesting - you seem to argue that there should be cross-language expected behavior for keywords. Would you say that's true for all elements of a PL?